File: src/test/resources/io/vertx/scala/core/buffer/Buffer.scala
package io.vertx.scala.core.buffer

class Buffer(delegate: io.vertx.core.buffer.Buffer) {

  def appendString(str: String) = {
    delegate.appendString(str);
    this
  }

  def toString(enc: String): String = {
    this.delegate.toString(enc);
  }
}

object Buffer {
  def buffer(string: String):Buffer = {
    new Buffer(io.vertx.core.buffer.Buffer.buffer(string))
  }
}

File: src/test/java/io/vertx/kotlin/core/buffer/Buffer.java
package io.vertx.kotlin.core.buffer;

/**
 * Created by Sergey Mashkov
 */
public class Buffer {
  private final io.vertx.core.buffer.Buffer delegate;

  public Buffer(io.vertx.core.buffer.Buffer delegate) {
    this.delegate = delegate;
  }

  public static Buffer buffer(String string) {
    return new Buffer(io.vertx.core.buffer.Buffer.buffer(string));
  }

  public Buffer appendString(String str) {
    delegate.appendString(str);
    return this;
  }

  public String toString(String enc) {
    return this.delegate.toString(enc);
  }
}
